gems/smithy/spec/fixtures/service_index/model.smithy

```diff
@@ -68,13 +72,38 @@ operation NestedResourceOperation {
     input: Structure
 }
 
+operation MapOperation {
+    input:= {
+        map: StructureMap
+    }
+}
+
 @documentation("This is a structure shape.")
 structure Structure {
     @required
     id: String
     name: String
 }
 
+list StructureList {
+    member: StructureMember
+}
+
+structure StructureMember {
+    id: String
+}
+
+map StructureMap {
+    key: MapKey
+    value: MapValue
+}
+
+string MapKey
+
+structure MapValue {
```

Comment:
Key and value could just be shapes that target strings for simplicity.

Reply:
See comment on list.  They must target shapes that are not referenced anywhere else.